### DISCUSSION TITLE
Is there any way to add Key Value Pair into existing languages at Runtime?

### DISCUSSION THREAD

**Discussion by gunpal5** [abpframework/abp#9508] [Q&A] [answered]
Hello,

I was wondering if there's anyway to add key-value pairs into existing Languages at runtime? So that admin users could add new Key/Token into existing languages and corresponding Values for different languages, and these runtime keys and values be delivered to clients along with the hard coded key value pairs.

Edit: 
e.g. lets say we add a new Token "App.ThisIsNewToken" into English Language and add it's value "This is a new Token". This newly created token should be added into the language dictionary in the "/api/abp/application-configuration" response.

Regards,
Gunpal Jain

**Comment by maliming** [comment]
HI

This feature available in abp commercial.
https://commercial.abp.io/modules/Volo.LanguageManagement

**Reply by gunpal5** [reply]
Yes, but you can't add new key value pairs in language, you can only create new languages and edit the existing hard coded tokens. 

![image](https://user-images.githubusercontent.com/10114874/124589109-3bc25c80-de77-11eb-9540-3e31d8b4a9b0.png)

**Comment by gunpal5** [comment, marked as answer]
I was digging into the localization module, I think this can be achieved by creating a new class for ILocalizationResourceContributor which populate the language dictionary from database instead of json files.

I'll try to do this.

**Reply by gunpal5** [reply]
Just Tried it and it worked. Here is a simple implementation based on MongoDb, but similar implementation can be done for other DB types or generalized code for all DB types:

https://gist.github.com/gunpal5/84beea419558519fe63ac3e5d04a119a
